components/email.php

```diff
@@ -0,0 +1,186 @@
+	 */
+	public function create( $args, $assoc_args ) {
+		$switched = false;
+
+		if ( false === bp_is_root_blog() ) {
+			$switched = true;
+			switch_to_blog( bp_get_root_blog_id() );
+		}
+
+		$term = term_exists( $assoc_args['type'], bp_get_email_tax_type() );
+
+		// Term already exists so don't do anything.
+		if ( $term !== 0 && $term !== null ) {
+			if ( true === $switched ) {
+				restore_current_blog();
+			}
+
+			WP_CLI::error( "Email type '{$assoc_args['type']}' already exists." );
+		}
+
+		if ( ! empty( $args[0] ) ) {
+			$assoc_args['content'] = $this->read_from_file_or_stdin( $args[0] );
+		}
+		if ( \WP_CLI\Utils\get_flag_value( $assoc_args, 'edit' ) ) {
+			$input = \WP_CLI\Utils\get_flag_value( $assoc_args, 'content', '' );
+			if ( $output = $this->_edit( $input, 'WP-CLI: New BP Email Content' ) ) {
+				$assoc_args['content'] = $output;
+			} else {
+				$assoc_args['content'] = $input;
+			}
+		}
+
+		$id = $assoc_args['type'];
+
+		$defaults = array(
+			'post_status' => 'publish',
+			'post_type'   => bp_get_email_post_type(),
+		);
+
+		$email = array(
+			'post_title'   => $assoc_args['subject'],
+			'post_content' => $assoc_args['content'],
+			'post_excerpt' => ! empty( $assoc_args['plain-text-content'] ) ? $assoc_args['plain-text-content'] :'',
+		);
+
+		// Email post content.
+		$post_id = wp_insert_post( bp_parse_args( $email, $defaults, 'install_email_' . $id ) );
+
+		// Save the situation.
+		if ( ! is_wp_error( $post_id ) ) {
+			$tt_ids = wp_set_object_terms( $post_id, $id, bp_get_email_tax_type() );
+
+			// Situation description.
+			if ( ! is_wp_error( $tt_ids ) && ! empty( $assoc_args['type-description'] ) ) {
+				$term = get_term_by( 'term_taxonomy_id', (int) $tt_ids[0], bp_get_email_tax_type() );
+				wp_update_term( (int) $term->term_id, bp_get_email_tax_type(), array(
+					'description' => $assoc_args['type-description'],
+				) );
+			}
+
+			if ( true === $switched ) {
+				restore_current_blog();
+			}
+
+			WP_CLI::success( "Email post created for type '{$assoc_args['type']}'." );
+		} else {
+			if ( true === $switched ) {
+				restore_current_blog();
+			}
+
+			WP_CLI::error( "There was a problem creating the email post for type '{$assoc_args['type']}'." );
+		}
+	}
+
+	/**
+	 * Get details for a post connected to an email type.
+	 *
+	 * ## OPTIONS
+	 *
+	 * <type>
+	 * : The email type to fetch the post details for.
+	 *
+	 * [--field=<field>]
+	 * : Instead of returning the whole post, returns the value of a single field.
+	 *
+	 * [--fields=<fields>]
+	 * : Limit the output to specific fields. Defaults to all fields.
+	 *
+	 * [--format=<format>]
+	 * : Render output in a particular format.
+	 * ---
+	 * default: table
+	 * options:
+	 *   - table
+	 *   - csv
+	 *   - json
+	 *   - yaml
+	 * ---
+	 *
+	 * ## EXAMPLE
+	 *
+	 *     # Output the post ID for the 'activity-at-message' email type
+	 *     $ wp bp email get-post activity-at-message --fields=ID
+	 *
+	 * @subcommand get-post
```

Comment:
Please, add also the `@alias see`. We are using this convention in almost all the `get` commands in wp-cli-bp.

Reply:
The naming `get_post` or `get-post` feels off to me - what about just `get`? Or maybe `get-type`?

Reply:
The rationale is there should be another command to grab the type term.  eg. `wp bp email get-type` would be just the email term and `wp bp email get-post` would be just for the email post object.

Let me know what you think here.

Reply:
That makes sense from a technical point of view, but practically, I wonder whether anyone would ever need to get the "term" separately. Let's err on the side of technical correctness: ignore my comment :-D

Reply:
The `see` alias is missing.